project_euler/problem_54/test_poker_hand.py

```diff
@@ -0,0 +1,234 @@
+]
+
+TEST_COMPARE = [
+    ("2H 3H 4H 5H 6H", "KS AS TS QS JS", "Loss"),
+    ("2H 3H 4H 5H 6H", "AS AD AC AH JD", "Win"),
+    ("AS AH 2H AD AC", "JS JD JC JH 3D", "Win"),
+    ("2S AH 2H AS AC", "JS JD JC JH AD", "Loss"),
+    ("2S AH 2H AS AC", "2H 3H 5H 6H 7H", "Win"),
+    ("AS 3S 4S 8S 2S", "2H 3H 5H 6H 7H", "Win"),
+    ("2H 3H 5H 6H 7H", "2S 3H 4H 5S 6C", "Win"),
+    ("2S 3H 4H 5S 6C", "3D 4C 5H 6H 2S", "Tie"),
+    ("2S 3H 4H 5S 6C", "AH AC 5H 6H AS", "Win"),
+    ("2S 2H 4H 5S 4C", "AH AC 5H 6H AS", "Loss"),
+    ("2S 2H 4H 5S 4C", "AH AC 5H 6H 7S", "Win"),
+    ("6S AD 7H 4S AS", "AH AC 5H 6H 7S", "Loss"),
+    ("2S AH 4H 5S KC", "AH AC 5H 6H 7S", "Loss"),
+    ("2S 3H 6H 7S 9C", "7H 3C TH 6H 9S", "Loss"),
+    ("4S 5H 6H TS AC", "3S 5H 6H TS AC", "Win"),
+    ("2S AH 4H 5S 6C", "AD 4C 5H 6H 2C", "Tie"),
+    ("AS AH 3H AD AC", "AS AH 2H AD AC", "Win"),
+    ("AH AC 5H 5C QS", "AH AC 5H 5C KS", "Loss"),
+    ("AH AC 5H 5C QS", "KH KC 5H 5C QS", "Win"),
+    ("7C 7S KH 2H 7H", "3C 3S AH 2H 3H", "Win"),
+    ("3C 3S AH 2H 3H", "7C 7S KH 2H 7H", "Loss"),
+    ("6H 5H 4H 3H 2H", "5H 4H 3H 2H AH", "Win"),
+    ("5H 4H 3H 2H AH", "5H 4H 3H 2H AH", "Tie"),
+    ("5H 4H 3H 2H AH", "6H 5H 4H 3H 2H", "Loss"),
+    ("AH AD KS KC AC", "AH KD KH AC KC", "Win"),
+    ("2H 4D 3C AS 5S", "2H 4D 3C 6S 5S", "Loss"),
+    ("2H 3S 3C 3H 2S", "3S 3C 2S 2H 2D", "Win"),
+    ("4D 6D 5D 2D JH", "3S 8S 3H TC KH", "Loss"),
+    ("4S 6C 8S 3S 7S", "AD KS 2D 7D 7C", "Loss"),
+    ("6S 4C 7H 8C 3H", "5H JC AH 9D 9C", "Loss"),
+    ("9D 9H JH TC QH", "3C 2S JS 5C 7H", "Win"),
+    ("2H TC 8S AD 9S", "4H TS 7H 2C 5C", "Win"),
+    ("9D 3S 2C 7S 7C", "JC TD 3C TC 9H", "Loss"),
+]
+
+TEST_FLUSH = [
+    ("2H 3H 4H 5H 6H", True),
+    ("AS AH 2H AD AC", False),
+    ("2H 3H 5H 6H 7H", True),
+    ("KS AS TS QS JS", True),
+    ("8H 9H QS JS TH", False),
+    ("AS 3S 4S 8S 2S", True),
+]
+
+TEST_STRAIGHT = [
+    ("2H 3H 4H 5H 6H", True),
+    ("AS AH 2H AD AC", False),
+    ("2H 3H 5H 6H 7H", False),
+    ("KS AS TS QS JS", True),
+    ("8H 9H QS JS TH", True),
+]
+
+TEST_FIVE_HIGH_STRAIGHT = [
+    ("2H 4D 3C AS 5S", True, [5, 4, 3, 2, 14]),
+    ("2H 5D 3C AS 5S", False, [14, 5, 5, 3, 2]),
+    ("JH QD KC AS TS", False, [14, 13, 12, 11, 10]),
+    ("9D 3S 2C 7S 7C", False, [9, 7, 7, 3, 2]),
+]
+
+TEST_KIND = [
+    ("JH AH TH KH QH", 0),
+    ("JH 9H TH KH QH", 0),
+    ("JC KH JS JD JH", 7),
+    ("KH KC 3S 3H 3D", 6),
+    ("8C 9C 5C 3C TC", 0),
+    ("JS QS 9H TS KH", 0),
+    ("7C 7S KH 2H 7H", 3),
+    ("3C KH 5D 5S KH", 2),
+    ("QH 8H KD JH 8S", 1),
+    ("2D 6D 9D TH 7D", 0),
+]
+
+TEST_TYPES = [
+    ("JH AH TH KH QH", 23),
+    ("JH 9H TH KH QH", 22),
+    ("JC KH JS JD JH", 21),
+    ("KH KC 3S 3H 3D", 20),
+    ("8C 9C 5C 3C TC", 19),
+    ("JS QS 9H TS KH", 18),
+    ("7C 7S KH 2H 7H", 17),
+    ("3C KH 5D 5S KH", 16),
+    ("QH 8H KD JH 8S", 15),
+    ("2D 6D 9D TH 7D", 14),
+]
+
+
+def generate_random_hands():
```

Comment:
```suggestion
def generate_random_hands(number_of_hands: int = 100):
    return (generate_random_hand() for _ in number_of_hands)
```

Reply:
Ah, the generator object, awesome idea! 😁
I will incorporate this and other changes you suggested right now.